ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OAutoML.scala

```diff
@@ -40,36 +46,95 @@ import scala.util.control.NoStackTrace
   * H2O AutoML algorithm exposed via Spark ML pipelines.
   */
 class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
-  with H2OAlgoCommonUtils with DefaultParamsWritable with H2OAutoMLParams {
+  with H2OAlgoCommonUtils with DefaultParamsWritable with H2OAutoMLParams with RestCommunication {
 
   // Override default values
   setDefault(nfolds, 5)
 
-  private lazy val spark = SparkSessionUtils.active
-
   def this() = this(Identifiable.randomUID(classOf[H2OAutoML].getSimpleName))
 
-  private var amlOption: Option[AutoML] = None
+  private var amlKeyOption: Option[String] = None
+
+  private def getInputSpec(trainKey: String, validKey: Option[String]): Map[String, Any] = {
+    Map(
+      "response_column" -> getLabelCol(),
+      "fold_column" -> getFoldCol(),
+      "weights_column" -> getWeightCol(),
+      "ignored_columns" -> getIgnoredCols(),
+      "sort_metric" -> getSortMetric(),
+      "training_frame" -> trainKey
+    ) ++ validKey.map{ key => Map("validation_frame" -> key) }.getOrElse(Map())
```

Comment:
I really like omitting space `map{` if the content is on the same line, but shouldn't we do this everywhere?

Reply:
Let's be consistent first, will fix to `map {`. 

After we introduce scala formatter we can decide about the right style